localstack-core/localstack/dev/kubernetes/__main__.py

```diff
@@ -1,56 +1,148 @@
-    volumes.append(
-        {
-            "volume": f"{os.path.normpath(localstack_code_path)}:/code/localstack",
-            "nodeFilters": ["server:*", "agent:*"],
-        }
-    )
+    pro_path = os.path.join(root_path, "..", "localstack-ext")
 
-    egg_path = os.path.join(
-        root_path, "localstack-core", "localstack_core.egg-info/entry_points.txt"
-    )
-    volumes.append(
-        {
-            "volume": f"{os.path.normpath(egg_path)}:/code/entry_points_community",
-            "nodeFilters": ["server:*", "agent:*"],
-        }
-    )
+    # container paths
+    target_path = "/opt/code/localstack/"
+    venv_path = os.path.join(target_path, ".venv", "lib", "python3.11", "site-packages")
+
+    # Community code
     if pro:
-        pro_path = os.path.join(root_path, "..", "localstack-ext")
-        pro_code_path = os.path.join(pro_path, "localstack-pro-core", "localstack", "pro", "core")
-        volumes.append(
-            {
-                "volume": f"{os.path.normpath(pro_code_path)}:/code/localstack_ext",
-                "nodeFilters": ["server:*", "agent:*"],
-            }
+        # Pro installs community code as a package, so it lives in the venv site-packages
+        mount_points.append(
+            MountPoint(
+                name="localstack",
+                host_path=os.path.normpath(localstack_code_path),
+                node_path="/code/localstack",
+                container_path=os.path.join(venv_path, "localstack"),
+                # Read only has to be false here, as we mount the pro code into this mount, as it is the entire namespace package
+                read_only=False,
+            )
         )
-
-        egg_path = os.path.join(
-            pro_path, "localstack-pro-core", "localstack_ext.egg-info/entry_points.txt"
+    else:
+        # Community does not install the localstack package in the venv, but has the code directly in `/opt/code/localstack`
+        mount_points.append(
+            MountPoint(
+                name="localstack",
+                host_path=os.path.normpath(localstack_code_path),
+                node_path="/code/localstack",
+                container_path=os.path.join(target_path, "localstack-core", "localstack"),
+            )
         )
-        volumes.append(
-            {
-                "volume": f"{os.path.normpath(egg_path)}:/code/entry_points_ext",
-                "nodeFilters": ["server:*", "agent:*"],
-            }
+
+    # Pro code
+    if pro:
+        pro_code_path = os.path.join(pro_path, "localstack-pro-core", "localstack", "pro", "core")
+        mount_points.append(
+            MountPoint(
+                name="localstack-pro",
+                host_path=os.path.normpath(pro_code_path),
+                node_path="/code/localstack-pro",
+                container_path=os.path.join(venv_path, "localstack", "pro", "core"),
+            )
         )
 
+    # entrypoints
+    if mount_entrypoints:
```

Comment:
Can we log a warning that this is currently not _really_ implemented, and may provide confusing results if the user expects this to work

Reply:
Also, note for the future: we implement this in `localstack.dev.run` so we can take the implementation from there.

See [`localstack.dev.run.configurators._list_files_in_container_image`](https://github.com/localstack/localstack/blob/ed8046a0d95762dceb94ac013671c1527d082afc/localstack-core/localstack/dev/run/configurators.py#L342)